csrc/device_lower/pass/scalar_hoist.cpp

```diff
@@ -213,6 +221,12 @@ std::pair<Val*, bool> CommonScalarMap::hoistScalarImpl(
     create_fn(value->container(), inputs, {value}, def->attributes());
   }
 
+  // TODO: this is only a temoprary hack, will be removed in a follow-up PR
```

Comment:
I need to write a more sophisticated approach for handling hoisting-to-host. Will likely do it after this PR.